dvc/stage/run.py

```diff
@@ -47,7 +47,7 @@ def warn_if_fish(executable):
 
 def _enforce_cmd_list(cmd):
     assert cmd
-    return cmd if isinstance(cmd, list) else [cmd]
+    return cmd if isinstance(cmd, list) else cmd.splitlines()
```

Comment:
A minor thing: What should we do in the `dvc.lock` file? Should we keep them in a single format (like either a list or a multiline?)? Just that when the user changes the multiline string to a list or vice versa, it should not affect the lock file.

It might be too much to support that way, but just saying that in principle, it'd be better to have a single format in the lock file, but I think it happens rarely in practice and the list and the multiline string might never be equivalent when user refactors them.

Reply:
@skshetry Great point! This PR still uses old schema where we save the cmd as is, but it makes total sense to just enforce the list in the new lock format. I guess we can do it in https://github.com/iterative/dvc/pull/5128 ?